ntoskrnl/mm/marea.c

```diff
@@ -331,13 +331,16 @@ MmFreeMemoryArea(
                 ASSERT(AddressSpace != MmGetKernelAddressSpace());
                 if (MiQueryPageTableReferences((PVOID)Address) == 0)
                 {
+                    KIRQL OldIrql;
                     /* No PTE relies on this PDE. Release it */
-                    KIRQL OldIrql = MiAcquirePfnLock();
+                    MiLockProcessWorkingSet(Process, PsGetCurrentThread());
+                    OldIrql = MiAcquirePfnLock();
                     PMMPDE PointerPde = MiAddressToPde(Address);
                     ASSERT(PointerPde->u.Hard.Valid == 1);
                     MiDeletePte(PointerPde, MiPdeToPte(PointerPde), Process, NULL);
```

Comment:
Might be worth annotating this function as requiring both locks. I'm aware that we are far away from being able to use /Analyze and mostly it's broken anyway, but who knows, it might change in the future. And for now it would serve as a quick documentation.

Reply:
Done